### PR TITLE
refactor: separate workload types in `cli/deploy`

### PR DESCRIPTION
**Pull request by dannyrandall**
no changes, just moving code to different files 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/deploy/job_test.go [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (efekarakus) [inline internal/pkg/cli/deploy/job_test.go:1]: should we delete these empty test files? :rip:

Comment (dannyrandall) [reply internal/pkg/cli/deploy/job_test.go:1]: i think it's kinda sus that they are empty, so i left them.... 🤔

Comment (dannyrandall) [reply internal/pkg/cli/deploy/job_test.go:1]: (as in, tests on `WorkloadDeployer` should probably be on these types)

File: internal/pkg/cli/deploy/rdws.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+func (rdwsDeployer) IsServiceAvailableInRegion(region string) (bool, error) {
+	return partitions.IsAvailableInRegion(awsapprunner.EndpointsID, region)
+}
+
+// UploadArtifacts uploads the deployment artifacts such as the container image, custom resources, addons and env files.
+func (d *rdwsDeployer) UploadArtifacts() (*UploadArtifactsOutput, error) {
+	return d.uploadArtifacts(d.customResources)
+}
+
+type rdwsDeployOutput struct {
```
Comment (Lou1415926) [inline internal/pkg/cli/deploy/rdws.go:79]: Can we move this struct definition and its `RecommendedAction` to just before `DeployWorkload` or to after all of `rdwsDeployer`'s public methods?

File: internal/pkg/cli/deploy/job.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	*GenerateCloudFormationTemplateOutput, error) {
+	output, err := d.stackConfiguration(&in.StackRuntimeConfiguration)
+	if err != nil {
+		return nil, err
+	}
+	return d.generateCloudFormationTemplate(output.conf)
+}
+
+// DeployWorkload deploys a job using CloudFormation.
+func (d *jobDeployer) DeployWorkload(in *DeployWorkloadInput) (ActionRecommender, error) {
```
Comment (Lou1415926) [inline internal/pkg/cli/deploy/job.go:69]: Is this method untested?

File: internal/pkg/cli/deploy/workload_test.go [hunk 4]
```diff
@@ -0,0 +1,1176 @@
...
+				fs:                 m.mockFileReader,
+				s3Client:           m.mockUploader,
+				imageBuilderPusher: m.mockImageBuilderPusher,
+				templateFS:         fakeTemplateFS(),
+			}
+			if m.mockAddons != nil {
+				wkldDeployer.addons = m.mockAddons
+			}
+			var deployer artifactsUploader
+			deployer = &lbWebSvcDeployer{
```
Comment (Lou1415926) [inline internal/pkg/cli/deploy/workload_test.go:416]: It feels to me that the unit tests here in `workload_test.go` should be testing constructing a workloadDeployer and calls its `uploadArtifacts` method for testing. But I guess it doesn't matter to much given that lbws's `UploadArtifacts` is virtually workloadDeployer's `uploadArtifacts`

**Review by Lou1415926 — APPROVED** [review]

**Review by iamhopaul123 — APPROVED** [review]

**Pull request merged**